lib/opt_struct/class_methods.rb

```diff
@@ -11,9 +11,11 @@ def required_keys
       @required_keys ||= []
     end
 
-    def required(*keys)
+    def required(*keys, **options)
```

Comment:
I'm not sure about `required` taking options. This was meant to be a convenience/helper/shorthand method while more advanced needs are better served by `option` which is the most configurable.

I can see how it would be useful to have a basket of attributes that are all required and private. I'm going to have to do more thinking but I definitely think if we are changing `required` to take `options` then `option_accessor`/`option_reader`/`option_writer` probably need to take options too.